### PR TITLE
swap - route picker

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes RNBW-3452 & TEAM2-92

## What changed (plus any additional context for devs)

users can now specify route's for swaps

also adds android route

## PoW (screenshots / screen recordings)
https://cloud.skylarbarrera.com/Screen-Recording-2022-06-07-14-37-53.mp4

https://user-images.githubusercontent.com/29204161/173709870-41382da0-ecf7-4171-a6cd-97fa0f25e765.mov

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/components/expanded-state/swap-settings/RoutePicker.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  );
+  const onPressAndroid = useCallback(() => {
+    showActionSheetWithOptions(
+      {
+        options: androidRouteMenuItems,
+        showSeparators: true,
+      },
+      idx => {
+        if (idx !== undefined) {
+          const routeOptions = androidRouteMenuItems();
```
Comment (estebanmino) [inline src/components/expanded-state/swap-settings/RoutePicker.js:60]: here the method is called

File: src/components/expanded-state/swap-settings/RoutePicker.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+              style={{
+                height: 20,
+                width: 20,
+              }}
+              width={20}
+            />
+            <Text size="18px" weight="bold">
+              {`${Object.keys(SwapRoute).find(
+                key => SwapRoute[key] === swapRoute
+              )} 􀆈`}
```
Comment (estebanmino) [inline src/components/expanded-state/swap-settings/RoutePicker.js:94]: i feel like this is too much to put it on a render, don't we have an object that we can just put the `swapRoute` and get a string out? 
also wondering about the `Both` word, makes sense in english but if we want to add additional languages this won't support it

File: src/components/expanded-state/swap-settings/RoutePicker.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+const androidRouteMenuItems = () => {
+  return Object.values(SwapRoute);
+};
+
+export default function SwapSettingsState({ onSelect, swapRoute }) {
+  const imageSource = useMemo(() => {
+    let source = null;
+    switch (swapRoute) {
+      case SwapRoute['1inch']:
```
Comment (brunobar79) [inline src/components/expanded-state/swap-settings/RoutePicker.js:32]: We should use  `Source.Aggregator0x`  and  `Source.Aggregotor1inch`

File: src/redux/swap.ts [hunk 4]
```diff
@@ -16,6 +16,12 @@ export enum SwapModalField {
   output = 'outputAmount',
 }
 
+export enum SwapRoute {
+  Both = 'Both',
+  '0x' = '0x',
```
Comment (brunobar79) [inline src/redux/swap.ts:21]: same  Source from sdk

**Review by brunobar79 — APPROVED** [review]

File: src/redux/swap.ts (hunk 4 above)
Comment (skylarbarrera) [reply src/redux/swap.ts]: fixed

File: src/components/expanded-state/swap-settings/RoutePicker.js (hunk 2 above)
Comment (skylarbarrera) [reply src/components/expanded-state/swap-settings/RoutePicker.js]: fixed

File: src/components/expanded-state/swap-settings/RoutePicker.js (hunk 3 above)
Comment (skylarbarrera) [reply src/components/expanded-state/swap-settings/RoutePicker.js]: fixed

**Review by estebanmino — APPROVED** [review]
🥇

**Pull request merged**